src/cluster.c

```diff
@@ -4948,6 +4962,163 @@ void addNodeReplyForClusterSlot(client *c, clusterNode *node, int start_slot, in
     setDeferredArrayLen(c, nested_replylen, nested_elements);
 }
 
+void addNodeDetailsToShardReply(client *c, clusterNode *node) {
+    /*
+     * Following information regarding a node is returned.
+     *
+     * 1. id (40 char len)
+     * 2. port
+     * 3. ip
+     * 4. hostname
+     * 5. endpoint
+     * 5. role
+     * 6. replication offset (in case of replica only).
+     * 7. health
+     */
+    int reply_cnt = 0;
+    void *node_replylen = addReplyDeferredLen(c);
+    addReplyBulkCString(c, "id");
+    addReplyBulkCBuffer(c, node->name, CLUSTER_NAMELEN);
+    reply_cnt++;
+    addReplyBulkCString(c, "port");
```

Comment:
I think we want to have TCP and TLS port here if available. We need to do some computation.

FYI @zuiderkwast

Reply:
Yes, I agree. Probably something like this:

```C
    int port = server.cluster_announce_port ?
        server.cluster_announce_port : server.port;
    int tls_port = server.cluster_announce_tls_port ?
        server.cluster_announce_tls_port : server.tls_port;
```

and omitting any port if it's zero.